src/test/java/org/mockito/internal/creation/instance/ConstructorInstantiatorTest.java

```diff
@@ -46,7 +46,7 @@ public void explains_when_constructor_cannot_be_found() {
             fail();
         } catch (InstantiationException e) {
             assertThat(e).hasMessageContaining("Unable to create instance of 'SomeClass2'.\n" +
-                    "Please ensure it has 0-arg constructor which invokes cleanly.");
+                    "Please ensure that it has a constructor that matches these arguments: []");
```

Comment:
I think that previous exception was a bit clearer. Let's keep the logic that presents cleaner exception even at the cost of a little bit of conditional complexity. In Mockito we strive to have super clean exception messages because that's where users need most help - when troubleshooting.